### PR TITLE
Splitted io test into each format

### LINKED ISSUE #2708: Splitting io test in `py-polars`
All of io test in `py-polars` in one file now, it is difficult to what tests are implemented in each format.

I think we should split the tests into `csv`, `ipc`, `json`, `parquet`, and `avro` like lazy_io tests.

### PR DESCRIPTION
**Pull request by illumination-k**
Splitting io test into each format.

## Finded Issues

I found the following bugs when rewriting. I will create issues for details later.

- got ValueError when writing ipc with compression (simple one is succeeded)
- lzo compression is not supported in arrow2 and pyarrow in parquet format
- got RuntimeError when writing json to file. fixed truncated issues (related to #2692), but I still got RuntimeError 

## Coverage

Master branch

```
----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                                                                                     Stmts   Miss  Cover
----------------------------------------------------------------------------------------------------------------------------
illumination-k/polars/py-polars/polars/__init__.py                      24      0   100%
illumination-k/polars/py-polars/polars/_html.py                         85      7    92%
illumination-k/polars/py-polars/polars/cfg.py                           32      0   100%
illumination-k/polars/py-polars/polars/convert.py                       33      2    94%
illumination-k/polars/py-polars/polars/datatypes.py                     80      1    99%
illumination-k/polars/py-polars/polars/datatypes_constructor.py         25      0   100%
illumination-k/polars/py-polars/polars/internals/__init__.py             8      0   100%
illumination-k/polars/py-polars/polars/internals/construction.py       201      3    99%
illumination-k/polars/py-polars/polars/internals/expr.py               601     24    96%
illumination-k/polars/py-polars/polars/internals/frame.py              931     87    91%
illumination-k/polars/py-polars/polars/internals/functions.py           59      3    95%
illumination-k/polars/py-polars/polars/internals/lazy_frame.py         307     62    80%
illumination-k/polars/py-polars/polars/internals/lazy_functions.py     350     50    86%
illumination-k/polars/py-polars/polars/internals/series.py             866     31    96%
illumination-k/polars/py-polars/polars/internals/whenthen.py            35      1    97%
illumination-k/polars/py-polars/polars/io.py                           190     51    73%
illumination-k/polars/py-polars/polars/string_cache.py                  15      0   100%
illumination-k/polars/py-polars/polars/testing.py                       51      1    98%
illumination-k/polars/py-polars/polars/utils.py                        100     23    77%
----------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                     3993    346    91%

Required test coverage of 91% reached. Total coverage: 91.33%
```

After splitted and added test

```
----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                                                                                     Stmts   Miss  Cover
----------------------------------------------------------------------------------------------------------------------------
illumination-k/polars/py-polars/polars/__init__.py                      24      0   100%
illumination-k/polars/py-polars/polars/_html.py                         85      7    92%
illumination-k/polars/py-polars/polars/cfg.py                           32      0   100%
illumination-k/polars/py-polars/polars/convert.py                       33      2    94%
illumination-k/polars/py-polars/polars/datatypes.py                     80      1    99%
illumination-k/polars/py-polars/polars/datatypes_constructor.py         25      0   100%
illumination-k/polars/py-polars/polars/internals/__init__.py             8      0   100%
illumination-k/polars/py-polars/polars/internals/construction.py       201      3    99%
illumination-k/polars/py-polars/polars/internals/expr.py               601     24    96%
illumination-k/polars/py-polars/polars/internals/frame.py              931     87    91%
illumination-k/polars/py-polars/polars/internals/functions.py           59      3    95%
illumination-k/polars/py-polars/polars/internals/lazy_frame.py         307     62    80%
illumination-k/polars/py-polars/polars/internals/lazy_functions.py     350     50    86%
illumination-k/polars/py-polars/polars/internals/series.py             866     31    96%
illumination-k/polars/py-polars/polars/internals/whenthen.py            35      1    97%
illumination-k/polars/py-polars/polars/io.py                           190     51    73%
illumination-k/polars/py-polars/polars/string_cache.py                  15      0   100%
illumination-k/polars/py-polars/polars/testing.py                       51      0   100%
illumination-k/polars/py-polars/polars/utils.py                        100     23    77%
----------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                     3993    345    91%

Required test coverage of 91% reached. Total coverage: 91.36%
```

close #2708

### CODE REVIEW

File: py-polars/tests/io/test_ipc.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+def test_from_to_buffer(df: pl.DataFrame, compressions: List[Compression]) -> None:
+    for compression in compressions:
+        if compression == "uncompressed":
+            buf = io.BytesIO()
+            df.to_ipc(buf, compression=compression)
+            buf.seek(0)
+            read_df = pl.read_ipc(buf)
+            assert df.frame_equal(read_df)
+        else:
+            # Error with ipc compression
```
Comment (illumination-k) [inline py-polars/tests/io/test_ipc.py:26]: Here is ipc compression error

File: py-polars/tests/io/test_ipc.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    expected = pl.DataFrame({"b": [True, False, True], "c": ["a", "b", "c"]})
+    f = io.BytesIO()
+    df.to_ipc(f)  # type: ignore
+    f.seek(0)
+
+    read_df = pl.read_ipc(f, columns=[1, 2], use_pyarrow=False)  # type: ignore
+    assert expected.frame_equal(read_df)
+
+
+def test_compressed_simple() -> None:
```
Comment (illumination-k) [inline py-polars/tests/io/test_ipc.py:74]: I can write very simple ipc even with compression

File: py-polars/tests/io/test_json.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    )
+    assert df.frame_equal(read_df)
+
+
+def test_to_from_file(io_test_dir: str, df: pl.DataFrame) -> None:
+    f = os.path.join(io_test_dir, "small.json")
+    df.to_json(f)
+
+    # Not sure why error occur
+    with pytest.raises(RuntimeError):
```
Comment (illumination-k) [inline py-polars/tests/io/test_json.py:25]: buffer works well, but a real file does not

**Comment by ritchie46** [comment]
Thanks a lot @illumination-k. The IPC error may be resolved by updating arrow. I will follow up with that.

**Pull request merged**